l1-contracts/contracts/state-transition/chain-deps/facets/Admin.sol

```diff
@@ -288,12 +288,24 @@ contract AdminFacet is ZkSyncHyperchainBase, IAdmin {
     }
 
     /// @inheritdoc IAdmin
-    function forwardedBridgeClaimFailedBurn(
-        uint256 _chainId,
-        bytes32 _assetInfo,
+    function forwardedBridgeRecoverFailedTransfer(
+        uint256 /* _chainId */,
+        bytes32 /* _assetInfo */,
         address _prevMsgSender,
-        bytes calldata _data
-    ) external payable override onlyBridgehub {}
+        bytes calldata _chainData
+    ) external payable override onlyBridgehub {
+        // As of now all we need in this function is the chainId so we encode it and pass it down in the _chainData field
+        uint256 protocolVersion = abi.decode(_chainData, (uint256));
+
+        require(s.settlementLayer != address(0), "Af: not migrated");
+        require(_prevMsgSender == s.admin, "Af: not chainAdmin");
```

Comment:
add a comment that this does not check that the caller of this function is the chain admin, but merely serves as a sanity check. 

as of now, any address can recover from failed transfer